File: loopr-ios/Models/Token/TokenType.swift
//
//  Depth.swift
//  loopr-ios
//
//  Created by kenshin on 2018/3/6.
//  Copyright © 2018年 Loopring. All rights reserved.
//

import Foundation

enum TokenType: CustomStringConvertible {

    case erc20 = "TOKEN_TYPE_ERC20"
    case erc1400 = "TOKEN_TYPE_ERC1400"
    case eth = "TOKEN_TYPE_ETH"
    case unknown = "UNKNOWN"

    var description: String {
        switch self {
        case .erc20:
            return "erc20"
        case .erc1400:
            return "erc1400"
        case .eth:
            return "eth"
        case .unknown:
            return "unknown"
        }
    }
}

File: walletsdk/src/main/java/leaf/prod/walletsdk/model/token/TokenType.java
package leaf.prod.walletsdk.model.token;

import com.google.gson.annotations.SerializedName;

/**
 * Created with IntelliJ IDEA.
 * User: kenshin wangchen@loopring.org
 * Time: 2019-03-21 2:37 PM
 * Cooperation: loopring.org 路印协议基金会
 */
public enum TokenType {

    @SerializedName("TOKEN_TYPE_ERC20")
    ERC20("ERC20"),

    @SerializedName("TOKEN_TYPE_ERC1400")
    ERC1400("ERC1400"),

    @SerializedName("TOKEN_TYPE_ETH")
    ETH("ETH"),

    UNKNOWN("UNKNOWN");

    private final String description;

    TokenType(final String text) {
        this.description = text;
    }

    public String getDescription() {
        return description;
    }
}
